### DISCUSSION TITLE
Possible to increase Peaqev allowed current

### DISCUSSION THREAD

**Discussion by rabakaas** [elden1337/hass-peaq#182] [Ideas]
Is it possible to increase Peaqev allowed current to over 16A?
Been using Peaqev for a while and its working great, but need to increase maximum allowed charging current to 32A on Easee charger.
I`m having two electric cars who needs to charge same day, and it would be great to charge a little faster sometimes and still keep my hourly consumption under 10Kwh

**Comment by elden1337** [comment]
Hi, thanks for using and glad you enjoy!

I am actually currently looking into something similar, but am a bit gnarly to allow >16A since im unsure of what would happen if i try and push more than allowed on a specific ciecuit (probably nothing but dont know yet). 

There are hidden lists of amps 16-32 already in the code. 

Do you know if easee shows somewhere in HA what the max allowed current is?

**Comment by rabakaas** [comment]
Easee reports max charger limit, 32A in my installation. sensor.ehwvXXXX_max_charger_limit
Any hints where I can find the hidden list in code to give it a try?

**Reply by elden1337** [reply]
It's in the core repo so you couldnt change it on the fly unfortunately.

**Comment by elden1337** [comment]
https://github.com/elden1337/hass-peaq/issues/183

**Comment by elden1337** [comment]
@rabakaas what is the state returned if i query that sensor? is it just a number?

**Comment by rabakaas** [comment]
Yes, just a number.

**Reply by elden1337** [reply]
https://github.com/elden1337/hass-peaq/releases/tag/v2.5.0-beta

try the beta if you want, would be appreciated. report back what works and not.

**Reply by rabakaas** [reply]
Thanks, charging with 22A now so seems to be working. Having some load from boiler and heatpump now, so 22A is correct. Will be testing a couple of days and report back.

**Reply by rabakaas** [reply]
Very promising so far..

![bilde](https://user-images.githubusercontent.com/94404198/207978602-b948a8c2-493e-469c-8958-5c0721e13adf.png)

**Comment by rabakaas** [comment]
This has been working great until today. Updated to latest Peaqev v2.5.0 and Easee v0.9.48 today and seems like Peaqev isnt reading  max_charger_limit. Tried with previous versions of both, but same problem.

![bilde](https://user-images.githubusercontent.com/94404198/208723881-dd963fd3-96e0-4fa3-a241-484979e39522.png)

![bilde](https://user-images.githubusercontent.com/94404198/208724050-5257c53c-c775-45dc-959c-8f54ec394bd1.png)

**Reply by elden1337** [reply]
Ajaj, it must be the easee-update. Will have to check. 

The 2.5.0 is the same as the beta you ve been running. 

What happens if you reboot? Still same?

**Comment by rabakaas** [comment]
Yes, same after reboot. Tried to downgrade easee, but still the same. MvhRobert20. des. 2022 kl. 18:20 skrev Magnus Eldén ***@***.***>:﻿
Ajaj, it must be the easee-update. Will have to check.
The 2.5.0 is the same as the beta you ve been running.
What happens if you reboot? Still same?

**Reply by elden1337** [reply]
Ok. Is this sensor showing up from easee at all in your HA now? Does it still just have a number showing the max amps?

If the sensors are showing up im confused.

Btw, did you go from beta2 och beta to 2.5.0? If you went from first beta there could be issues i've caused.

**Reply by elden1337** [reply]
Here is the diff between first beta and current version. from beta.2 the diff is even smaller. 
https://github.com/elden1337/hass-peaq/compare/v2.5.0-beta...v2.5.0

So my hope is that there is something going on with the easee integration, perhaps an api-issue on their side that prevents peaqev from reading the sensor.
Please get back if the error persists. I cannot see anything that I could change to make it work (and this is unfortunately quite often the story with Easee)

**Reply by elden1337** [reply]
I did check some more and deeper now. And this second warning actually means that peaqev cannot communicate with the charger-integration it is set to talk to, thus it is not initialized properly or it's api is down so peaqev cannot get the state. 
I've seen it before, and it does come back when easee is working well again. 

I'm going to change that warning going forward to: "Unable to communicate with the charger-integration. Chargerobject value is: {self.value}. Unable to continue. Please check and reboot Home Assistant."

**Comment by rabakaas** [comment]
Sorry for late reply, busy in a meeting..
Uninstalled Peaqev and did a fresh install, V2.5.0,  working now.
Will do more testing tomorrow to see if it survives a restart of HA.
